### PR TITLE
THU/limit_get_embed_info_requested_reports

### PR DESCRIPTION
**Pull request by csm-thu**
- refactor/cleaning in api folder
  - remove unused function `getEmbedParamsForMultipleReports` in _api/GetEmbedInfo/embedConfigService.js_
  - remove unused file _api/GetEmbedInfo/sample.dat_
- add `reportsIds` parameter for the Azure Function "get-embed-info", the query is now a POST request instead of GET
- move some logic into _src/services/powerbi/PowerBIService.js_ to simplify the PowerBI redux saga

### CODE REVIEW

File: src/utils/PowerBIUtils.js [hunk 1]
```diff
@@ -51,7 +51,23 @@ const getScenarioViewReportConfig = (reportsConfig, runTemplateId) => {
...
+    return reportsConfig.map((reportConfig) => reportsConfig.map((report) => report.reportId));
+  }
+  return Object.values(reportsConfig).map((report) => report.reportId);
+};
+
+const getReportsIdsFromConfig = (config) => {
+  const reportsIds = _getReportsIdsFromDashboardsViewConfig(config.dashboardsView).concat(
+    _getReportsIdsFromScenarioViewConfig(config.scenarioView)
+  );
+  return reportsIds.filter((reportId, pos) => reportsIds.indexOf(reportId) === pos);
```
Comment (nborde-CSM) [inline src/utils/PowerBIUtils.js:66]: With this expression, you want to remove duplicate, indeed ?
you can do it easier like that :
`return Array.from(new Set (_getReportsIdsFromDashboardsViewConfig(config.dashboardsView),
    _getReportsIdsFromScenarioViewConfig(config.scenarioView));`

**Review by nborde-CSM — APPROVED** [review]
Two remarks but LGTM

File: src/utils/PowerBIUtils.js (hunk 1 above)
Comment (csm-thu) [reply src/utils/PowerBIUtils.js:66]: :+1:  Replaced code by the following:
```
  return Array.from(
    new Set([
      ..._getReportsIdsFromDashboardsViewConfig(config.dashboardsView),
      ..._getReportsIdsFromScenarioViewConfig(config.scenarioView),
    ])
  );
  ```

File: src/utils/PowerBIUtils.js [hunk 2]
```diff
@@ -51,7 +51,25 @@ const getScenarioViewReportConfig = (reportsConfig, runTemplateId) => {
     : [];
 };
 
+const _getReportsIdsFromDashboardsViewConfig = (reportsConfig) => reportsConfig.map((report) => report.reportId);
+const _getReportsIdsFromScenarioViewConfig = (reportsConfig) => {
+  if (Array.isArray(reportsConfig)) {
+    return reportsConfig.map((reportConfig) => reportsConfig.map((report) => report.reportId));
```
Comment (esasova) [inline src/utils/PowerBIUtils.js:57]: I don't see where is used `reportConfig` parameter ? :thinking: Why do we need a double loop to retrieve `reportId` ?

**Review by esasova — APPROVED** [review]
LGTM :+1:

File: api/GetEmbedInfo/embedConfigService.js [hunk 3]
```diff
@@ -41,125 +41,44 @@ async function getEmbedInfo() {
...
-    datasetIds.add(reportInfo.datasetId);
-    reportIds.add(reportInfo.id);
-
-    // Append to existing list of datasets to achieve dynamic binding later
-    if (additionalDatasetIds) {
-      datasetIds.add(...additionalDatasetIds);
-    }
+  for (const report of selectedReports) {
+    const reportDetails = new PowerBiReportDetails(report.id, report.name, report.embedUrl);
+    reportEmbedConfig.reportsDetail[report.id] = reportDetails;
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply api/GetEmbedInfo/embedConfigService.js:71]: These lines already existed before my changes, but the refactoring could be a good occasion to improve them :+1:

File: src/utils/PowerBIUtils.js (hunk 2 above)
Comment (csm-thu) [reply src/utils/PowerBIUtils.js:57]: Nice catch, copy/paste mistake :sweat_smile:

**Review by ardvi — APPROVED** [review]
LGTM

**Pull request merged**